src/App.js

```diff
@@ -1,39 +1,64 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
+import { Route, Redirect } from "react-router-dom";
 
 import UserProfile from "./components/User/UserProfile";
 import Photos from "./components/gallery/Photos";
+import Picture from "./components/Picture";
 import "./App.css";
 
 function App() {
-
   const [data, setData] = useState(null);
 
   useEffect(() => {
-     
-      let myHeaders = new Headers();
-      myHeaders.append("Origin", "https://localhost.com");
-      
-      let requestOptions = {
-        method: 'GET',
-        headers: myHeaders,
-        redirect: 'follow'
-      };
-
-      fetch("https://image-gallery-pjks.s3.ca-central-1.amazonaws.com/data.json", requestOptions)
-       .then(response => response.json())
-       .catch(() => console.log('add backup data'))
-       .then((response) => setData(response)); 
+    let myHeaders = new Headers();
+    myHeaders.append("Origin", "https://localhost.com");
+
+    let requestOptions = {
+      method: "GET",
+      headers: myHeaders,
+      redirect: "follow",
+    };
 
+    fetch(
+      "https://image-gallery-pjks.s3.ca-central-1.amazonaws.com/data.json",
+      requestOptions
+    )
+      .then((response) => response.json())
+      .catch(() => console.log("add backup data"))
+      .then((response) => setData(response));
   }, []);
 
   return (
-    <div className="App">
-      <UserProfile />
+    <React.Fragment>
+
+      <Route path='/PhotoGallery-V2' exact> 
+        <Redirect to='gallery' />
+      </Route>
+
+      <Route path='/gallery'>
+
+        <div className="App">
+          <UserProfile />
+
+          {data &&
+            data.map((image) => (
+              <Photos
+                source={image.url}
+                location={image.location}
+                description={image.descreption}
+                alt={`Photo-${image.id}`}
+                id = {image.id}
+                key={Math.random()}
+              />
+            ))}
+        </div>
+      </Route>
+
+      <Route path='/picture/:id'>
+         <Picture />
+      </Route>
```

Comment:
very good decision, im happy thats the route you decided to make. That is EXACTLY how you want to build navigation around ur site, e.g. with a different route, passing in path param (picture id), etc.